src/network/networkctl.c

```diff
@@ -2604,6 +2606,34 @@ static int lldp_neighbors_varlink_reply(Varlink *link, JsonVariant *parameters,
         udata->neighbors_count += 1;
         udata->capabilities_all |= entry.capabilities;
 
+        if (udata->json) {
+                _cleanup_(json_variant_unrefp) JsonVariant *v = NULL;
+                _cleanup_(json_variant_unrefp) JsonVariant *neighbors = NULL;
+
+                capabilities = lldp_capabilities_to_string(entry.capabilities, false);
+
+                r = json_build(&v, JSON_BUILD_OBJECT(
+                                        JSON_BUILD_PAIR("neighbor", JSON_BUILD_OBJECT(
+                                        JSON_BUILD_PAIR_CONDITION(entry.chassis_id, "chassisId", JSON_BUILD_STRING(entry.chassis_id)),
+                                        JSON_BUILD_PAIR_CONDITION(entry.port_id, "portId", JSON_BUILD_STRING(entry.port_id)),
+                                        JSON_BUILD_PAIR_CONDITION(entry.system_name, "systemName", JSON_BUILD_STRING(entry.system_name)),
+                                        JSON_BUILD_PAIR_CONDITION(entry.port_description, "portDescription", JSON_BUILD_STRING(entry.port_description)),
+                                        JSON_BUILD_PAIR_CONDITION(entry.capabilities, "enabledCapabilities", JSON_BUILD_STRING(capabilities))))));
```

Comment:
hmm, why not just add the literal json object from the server here? why remarshal?

Reply:
The JSON from the server does not contain capabilities as a string but as an unsigned (server encodes them as an unsigned and `networkctl` decodes them into a string you can see in `networkctl lldp` output even as of now). I was remarshalling here to decode the integer.

I agree that this might not be ideal so:
1) it might be better for the server to pass the array of strings directly and only pass the json here (perhaps even instead of string like `"bwr"` it might make sense to pass more verbose `["bridge", "wlan", "router"]`?)
2) only replace the one field in the variant (unsigned into string)?